### ISSUE TITLE
Global progress tracker

### ISSUE THREAD

**Issue by fchollet** [keras-team/keras#18474]
- [ ] Engine
  - [x] Step fusing [@haifeng-jin]
  - [ ] TPU support
  - [x] tf.distribute support
  - [ ] DTensor support
  - [ ] Sparse inputs support
  - [x] Stack trace filtering [@fchollet]
  - [x] Support for add_loss in stateless calls [@fchollet]
- [x] Losses
- [x] Metrics
- [x] Initializers
- [x] Regularizers
- [x] Constraints
- [x] Datasets
- [x] Data loading utils
- [x] Utils
- [ ] Layers
- [x] Preprocessing layers
- [ ] Callbacks
- [x] Optimizers
- [x] Learning rate schedules
- [ ] Data adapters
  - [x] Sequence/PyDataset and corresponding data adapters
  - [x] Python generator data adapter
  - [x] Pytorch data adapter
- [ ] Full test coverage
- [ ] Integration tests
- [ ] Backwards compatibility tests
- [ ] Benchmarks
- [ ] Website
- [ ] Comms
